roles/pulp_webserver/tasks/generate_tls_certificates.yml

```diff
@@ -0,0 +1,69 @@
+---
+- name: Set python-cryptography package name
+  set_fact:
+    python_cryptography: python-cryptography
+  when: ansible_python.version.major == 2
+
+- name: Set python-cryptography package name
+  set_fact:
+    python_cryptography: python3-cryptography
+  when: ansible_python.version.major == 3
+
+- name: Ensure python-cryptography is installed
+  become: true
+  package:
+    name: '{{ python_cryptography }}'
+
+- name: Generate CA key
+  openssl_privatekey:
+    path: '{{ pulp_webserver_tls_folder }}/root.key'
```

Comment:
I'm concerned about  what happens with these modules if a user runs the installer, replaces the key files on disk in the same filepaths with their desired keys, and then re-runs the installer:
https://docs.ansible.com/ansible/latest/modules/openssl_privatekey_module.html
> Please note that the module regenerates private keys if they don’t match the module’s options. In particular, if you provide another passphrase (or specify none), change the keysize, etc., the private key will be regenerated. If you are concerned that this could overwrite your private key, consider using the backup option.

Reply:
In a general approach, I would say don't do that. If this is what you want to do, use the installer to do it and re-run the installer with `pulp_webserver_ssl_cert` and `pulp_webserver_ssl_key`. If you don't yet says you want TLS then yes what you have described will happen. But I feel like it is doing it by design, rightfully.

If you think this is wrong. We can condition this whole block into the conditional non-existence off those 2 files.

Reply:
I agree. If the keys are managed by ansible, do not even think about messing around by hand.